pages/es/supported-networks/cosmos.mdx

```diff
@@ -0,0 +1,191 @@
+        - event: rewards # the type of the event that will be handled
+          handler: handleReward # the function name in the mapping file
+      transactionHandlers:
+        - handler: handleTransaction # the function name in the mapping file
+      file: ./src/mapping.ts # link to the file with the Assemblyscript mappings
+```
+
+- Cosmos Hub subgraphs introduce a new `kind` of data source (`cosmos`).
+- The `network` should correspond to a network on the hosting Graph Node. Cosmos Hub mainnet is `cosmoshub-4`. Other Cosmos Hub networks, i.e. `cosmoshub-3`, are halted, therefore no data is provided for them.
+
+Cosmos Hub data sources support three types of handlers:
+
+- `blockHandlers`: run on every new block appended to the chain. The handler will receive a full block and all its data containing, among other things, all the events and transactions.
+- `eventHandlers`: run on every event, except if there is an event type specified in the handler definition, in that case only events of that type are processed. Block data is also passed onto the mapping in order to have context of the event within the chain.
+- `transactionHandlers`: run for every transaction executed. The mapping is provided with all the relevant data related to the transaction and a block abstraction that can be used to acquire context of the transaction within a block and within the chain.
+
+Event and Transaction handlers are a way to process meaningful data from the chain without the need of processing a whole block. The data processed by them can also be found in the block handlers, since events and transactions are also part of a block, but removes the need of processing unnecessary data.
+
+### Schema Definition
+
+Schema definition describes the structure of the resulting subgraph database, and the relationships between entities. This is agnostic of the original data source. There are more details on subgraph schema definition [here](https://thegraph.com/docs/en/developer/create-subgraph-hosted/#the-graph-ql-schema).
+
+### AssemblyScript Mappings
+
+The handlers for processing events are written in [AssemblyScript](https://www.assemblyscript.org/).
+
+Cosmos Hub indexing introduces Cosmos-specific data types to the [AssemblyScript API](https://thegraph.com/docs/en/developer/assemblyscript-api/).
+
+```tsx
+class EventList {
+    newBlock: EventBlock
+    transaction: Array<EventTx>
+    validatorSetUpdates: EventValidatorSetUpdates
+}
+
+class EventData {
+    event: Event
+    block: EventBlock
+}
+
+class TransactionData {
+    tx: TxResult
+    block: EventBlock
+}
+
+class EventBlock {
+    block: Block
+    blockId: BlockID
+    resultBeginBlock: ResponseBeginBlock
+    resultEndBlock: ResponseEndBlock
+}
+
+class EventTx {
+		txResult: TxResult
+}
+
+class TxResult {
+		height: u64
+    index: u32
+    tx: Bytes
+    result: ResponseDeliverTx
+    hash: Bytes
+}
+
+class Event {
+    eventType: string
+    attributes: Array<EventAttribute>
```

Comment:
What is the structure of `EventAttribute`?

Reply:
I just wanted to highlight the most used ones. I reference the full list below the code snippet: https://github.com/graphprotocol/graph-ts/blob/main/chain/tendermint.ts

Should I add them all to the docs ?